commit 53f6278d8a655622843be842b9f0e50f70841af4
Author: Kenneth Hoste <kenneth.hoste@ugent.be>
Date:   Mon Oct 26 10:49:32 2020 +0100

    also compile Lmod spider cache for alternate Lua versions if corresponding luac command is available

diff --git a/src/update_lmod_system_cache_files.in b/src/update_lmod_system_cache_files.in
index e3a089ae..5daf8127 100644
--- a/src/update_lmod_system_cache_files.in
+++ b/src/update_lmod_system_cache_files.in
@@ -308,6 +308,18 @@ update_cache() {
        lua_ver=$(@path_to_lua@/lua -e 'print((_VERSION:gsub("Lua ","")))')
        @path_to_luac@ -o ${cache_file_name}.new.luac_$lua_ver ${cache_file_name}.lua
        install_new_cache luac_$lua_ver ${cache_file_name}
+
+       # also compile cache with other Lua 5.x versions, if corresponding luac is available
+       for alt_luac in $(ls @path_to_luac@-5.*); do
+           alt_lua_ver=$(basename ${alt_luac} | cut -f2 -d'-')
+           debug "Found potential alternate Lua version ${alt_lua_ver} (via ${alt_luac}"
+           # only for Lua versions different than what lua/luac commands correspond to (see above)
+           if [ "${alt_lua_ver}" != "${lua_ver}" ]; then
+               debug "Compiling cache file ${cache_file_name} for Lua ${alt_lua_ver}..."
+               ${alt_luac} -o ${cache_file_name}.new.luac_${alt_lua_ver} ${cache_file_name}.lua
+               install_new_cache luac_${alt_lua_ver} ${cache_file_name}
+           fi
+       done
     fi
 }
 

